aa-sdk/core/src/middleware/erc7677middleware.ts

```diff
@@ -159,21 +185,143 @@ export function erc7677Middleware<
 
   const paymasterAndData: ClientMiddlewareFn = async (
     uo,
-    { client, account, feeOptions, overrides }
+    { client, account, feeOptions, overrides, policyToken }
   ) => {
     const userOp = deepHexlify(await resolveProperties(uo));
-    const context =
+
+    let context =
       (typeof params?.context === "function"
         ? await params?.context(userOp, { overrides, feeOptions })
         : params?.context) ?? {};
+    const entrypoint = account.getEntryPoint();
 
     if (!client.chain) {
       throw new ChainNotFoundError();
     }
 
+    let max_token: BigInt | undefined = undefined;
+    if (policyToken !== undefined) {
+      let total_gas = BigInt(0);
+      total_gas += uo.callGasLimit
+        ? BigInt(uo.callGasLimit.toLocaleString())
```

Comment:
This doesn't work. `uo.callGasLimit` has type `PromiseOrValue<...>` meaning it is either an actual value or a promise that eventually resolves to a value. In the latter case, `.toLocaleString()` returns `"[object Promise]"` rather than anything representing the value.

You should read from `userOp` instead of `uo`, which represents the same operation with all fields awaited. Likewise throughout.